doc/mgmt/config_replace_rollback_HLD.md

```diff
@@ -0,0 +1,415 @@
+##### 3.2.2.2.1 Replace Command
+| replace command                                         | replace command inside configuration session to replace running configuration  |
+| ------------------------------------------------------- | ------------------------------------------------------------ |
+| **Syntax**                                              | replace source-url  |
+| **Parameters**                                          |
+| source-url                                              | config://filename<br>home://filename<br>ftp://[userid[:passwd]@]{hostname \| host-ip}/file-path<br>http[s]://[userid[:passwd]@]{hostname \| host-ip}/file-path<br>scp://[userid[:passwd]@]{hostname \| host-ip}/file-path|
+| **Default**                                             |  |
+| **Command mode**                                        | Config-s |
+| **Roles**                                               | admin |
+| **Example**                                             | Sonic(config-s)#replace http://inter128.acme.com/sample.json<br>Sonic(config-s)# |
+| **Support**                                             | 4.1 |
+##### 3.2.2.2.2 commit command
+| commit command                                          | commit command inside configuration session to kick start the timer or confirm the configuration changes |
+| ------------------------------------------------------- | ------------------------------------------------------------ |
+| **Syntax**                                              | commit [label \<LABEL\>] [time seconds] \| [confirm]  |
+| **Parameters**                                          |
+| confirm						  | To make the configuration changes permanent in running-configuration |
+| time   						  | 30 .. 3600 seconds for which the configuration changes kept in running-configuration as trial stage, default - 30 |
+| label   						  | commit label string, upto 72 characters |
+| **Default**                                             | None |
+| **Command mode**                                        | Config-s |
+| **Roles**                                               | admin |
+| **Example**                                             | Sonic(config-s)#commit time 120<br>Sonic(config-s)#commit confirm |
+| **Support**                                             | 4.1 |
+Note: "commit" will be implemented by Broadcom. time and confirm options will be added to "commit" command
+### 3.2.3 REST API Support
+Regular RPC operation can be sent from REST. The server side implementation should be the same as exce mode CLI.
+
+### 3.2.4 gNMI Support
+
+
+# 4 Flow Diagrams
+# 4.1 Exec mode configuration replace command
+![Exec_mode_replace](config_replace_images/Exec_mode.png "Figure 1: Exec mode replace")
+	
+ __Figure 1: Exec mode configuration replace__
+
+# 4.2 Config session mode replace command
+![Config_session_mode_replace](config_replace_images/Config_session_replace.png "Figure 2: Config Session replace")
```

Comment:
This also needs to talk to DB layer to commit to candidate configuration else I am not sure how the diff is going to work. Also there is no explanation on how the json diff will be achieved between the running configuration and the candidate configuration.

Reply:
we may need more clarity on the design and the flow with respect to computing the diff, and ordering the each entries based on its leafref dependency

Reply:
We do no use the json diff. Instead, we use map based comparison for tables, keys and fields after unmarshalling both running and intended configs. Purely programming comparison to do the intended job.